staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -140,6 +141,16 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		clientConfig.Timeout = timeout
 	}
 
+	if configClusterInfo.ConnectionTimeout != "" {
+		connectionTimeout, err := ParseTimeout(configClusterInfo.ConnectionTimeout)
+		if err != nil {
+			return nil, err
+		}
+		clientConfig.ConnectionTimeout = connectionTimeout
+	} else {
+		clientConfig.ConnectionTimeout = 7 * time.Second
```

Comment:
This is too blunt a hammer to be applied to all operations - it really is only appropriate for the specific client used for checkpointing to the master.  I would recommend finding a clean way to specifically target *only* the client that is used by node heart beating with this change.  That may require that the node create a specific Go client for that.